drivers/sensor/tmp117/tmp117.c

```diff
@@ -0,0 +1,189 @@
+			    uint16_t val)
+{
+	struct tmp117_data *drv_data = dev->data;
+	const struct tmp117_dev_config *cfg = dev->config;
+	uint8_t tx_buf[3] = {reg, val >> 8, val & 0xFF};
+
+	return i2c_write(drv_data->i2c, tx_buf, sizeof(tx_buf),
+			cfg->i2c_addr);
+}
+
+
+/**
+ * @brief Check the Device ID
+ *
+ * @param[in]   dev     Pointer to the device structure
+ *
+ * @retval 0 On success
+ * @retval -EIO Otherwise
+ */
+static inline int tmp117_device_id_check(const struct device *dev)
+{
+	uint16_t value;
+
+	if (tmp117_reg_read(dev, TMP117_REG_DEVICE_ID, &value) != 0) {
+		LOG_ERR("%s: Failed to get Device ID register!",
+			DT_INST_LABEL(0));
+		return -EIO;
+	}
+
+	if (value != TMP117_DEVICE_ID) {
+		LOG_ERR("%s: Failed to match the device IDs!",
+			DT_INST_LABEL(0));
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int tmp117_sample_fetch(const struct device *dev,
+			       enum sensor_channel chan)
+{
+	struct tmp117_data *drv_data = dev->data;
+	uint16_t value;
+	int rc;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL ||
+			chan == SENSOR_CHAN_AMBIENT_TEMP);
+
+	/* clear sensor values */
+	drv_data->sample = 0U;
+
+	/* Get the most recent temperature measurement */
+	rc = tmp117_reg_read(dev, TMP117_REG_TEMP, &value);
+	if (rc < 0) {
+		LOG_ERR("%s: Failed to read from TEMP register!",
+			DT_INST_LABEL(0));
+		return rc;
+	}
+
+	/* store measurements to the driver */
+	drv_data->sample = (int16_t)value;
+
+	return 0;
+}
+
+static int tmp117_channel_get(const struct device *dev,
+			      enum sensor_channel chan,
+			      struct sensor_value *val)
+{
+	struct tmp117_data *drv_data = dev->data;
+	int32_t tmp;
+
+	if (chan != SENSOR_CHAN_AMBIENT_TEMP) {
+		return -ENOTSUP;
+	}
+
+	/*
+	 * See datasheet "Temperature Results and Limits" section for more
+	 * details on processing sample data.
+	 */
+	tmp = ((int16_t)drv_data->sample * (int32_t)TMP117_RESOLUTION) / 10;
+	val->val1 = tmp / 1000000; /* uCelsius */
+	val->val2 = tmp % 1000000;
+
+	return 0;
+}
+
+static int tmp117_attr_set(const struct device *dev,
+			   enum sensor_channel chan,
+			   enum sensor_attribute attr,
+			   const struct sensor_value *val)
+{
+	/*
+	 * The offset is encoded into the temperature register format.
+	 */
+	int16_t value = (((val->val1) * 10000000) + ((val->val2) * 1000000))
+						/ (int32_t)TMP117_RESOLUTION;
+
+	if (chan != SENSOR_CHAN_AMBIENT_TEMP) {
+		return -ENOTSUP;
+	}
+
+	switch (attr) {
+	case SENSOR_ATTR_OFFSET:
+		return tmp117_reg_write(dev, TMP117_REG_Temp_OFFSET, value);
+
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static const struct sensor_driver_api tmp117_driver_api = {
+	.attr_set = tmp117_attr_set,
+	.sample_fetch = tmp117_sample_fetch,
+	.channel_get = tmp117_channel_get
+};
+
+static int tmp117_init(const struct device *dev)
+{
+	struct tmp117_data *drv_data = dev->data;
+	int rc;
+
+	/* Bind to the I2C bus that the sensor is connected */
+	drv_data->i2c = device_get_binding(DT_INST_BUS_LABEL(0));
```

Comment:
Would be nice to save `DT_INST_BUS_LABEL(0)` into `struct tmp117_dev_config tmp117_config `, same as `i2c_addr` is.